fdbclient/MonitorLeader.actor.cpp

```diff
@@ -601,8 +601,15 @@ ACTOR Future<Void> getClientInfoFromLeader( Reference<AsyncVar<Optional<ClusterC
 		}
 		req.knownClientInfoID = clientData->clientInfo->get().id;
 		choose {
-			when( ClientDBInfo ni = wait( brokenPromiseToNever( knownLeader->get().get().clientInterface.openDatabase.getReply( req ) ) ) ) {
+			when( state ClientDBInfo ni = wait( brokenPromiseToNever( knownLeader->get().get().clientInterface.openDatabase.getReply( req ) ) ) ) {
 				TraceEvent("MonitorLeaderForProxiesGotClientInfo", knownLeader->get().get().clientInterface.id()).detail("Proxy0", ni.proxies.size() ? ni.proxies[0].id() : UID()).detail("ClientID", ni.id);
+				if(ni.proxies.size() > CLIENT_KNOBS->MAX_CLIENT_PROXY_CONNECTIONS) {
+					deterministicRandom()->randomShuffle(ni.proxies);
```

Comment:
Since we are mutating the list of proxies here, [monitorMasterProxiesChanged](https://github.com/apple/foundationdb/blob/master/fdbclient/NativeAPI.actor.cpp#L461) will probably get triggered even when it shouldn't. Don't think it should matter much though, but that may also mean that ACTOR is probably unnecessary as it seems that Future is effectively `clientDBInfo->onChange()` as we'll shuffle this everytime which would make that ACTOR trigger onMasterProxyChanged() each time.